src/django_components/component_registry.py

```diff
@@ -268,6 +270,39 @@ def clear(self) -> None:
         self._registry = {}
         self._tags = {}
 
+    def _register_to_library(
+        self,
+        comp_name: str,
+        component: Type["Component"],
+    ) -> ComponentRegistryEntry:
+        # Lazily import to avoid circular dependencies
+        from django_components.templatetags.component_tags import do_component_tag_block, do_component_tag_inline
+
+        formatter = get_component_tag_formatter()
+
+        # TODO - Allow None here?
+        # register the inline tag
+        inline_tag = formatter.safe_format_inline_tag(comp_name)
+
+        if is_tag_protected(self.library, inline_tag):
+            raise TagProtectedError('Cannot register inline tag "%s", this tag name is protected' % inline_tag)
```

Comment:
Here we throw if user tried to register a component tag with one of the protected names.